src/Uno.UWP/Collections/MaterializableList.cs

```diff
@@ -47,6 +50,20 @@ public MaterializableList(IEnumerable<T> collection)
 
 		public MaterializableList<T>.ReverseEnumerator GetReverseEnumerator() => new ReverseEnumerator(Materialized);
 
+		/// <remarks>
+		/// To optimize sequential calls with the same sorting function, the implementation remembers the sorted list and the last
+		/// provided keySelector and compares it by reference. If it's the same function, the sorted list is reused.
+		/// </remarks>
+		public MaterializableList<T>.ReverseEnumerator GetReverseSortedEnumerator<TKey>(Func<T, TKey> keySelector)
+		{
+			if (_materializedSorted is null || ReferenceEquals(_lastSortingFunc, keySelector))
```

Comment:
`ReferenceEquals(_lastSortingFunc, keySelector)` your comparing a WeakRef with a delagate, the cache will never match